clarity/src/vm/ast/sugar_expander/mod.rs

```diff
@@ -509,4 +556,309 @@ mod test {
             "Should match expected symbolic expression"
         );
     }
+
+    #[test]
+    fn test_attach_end_line_comment() {
+        let pre_ast = vec![
+            make_pre_atom("foo", 1, 1, 1, 3),
+            make_pre_comment("this is a comment".to_string(), 1, 5, 1, 21),
+            make_pre_atom("bar", 2, 1, 2, 3),
+        ];
+        let mut foo = make_atom("foo", 1, 1, 1, 3);
+        foo.end_line_comment = Some("this is a comment".to_string());
+        let bar = make_atom("bar", 2, 1, 2, 3);
+        let ast = vec![foo, bar];
+
+        let contract_id = QualifiedContractIdentifier::parse(
+            "S1G2081040G2081040G2081040G208105NK8PE5.test-comments",
+        )
+        .unwrap();
+        let mut contract_ast = ContractAST::new(contract_id.clone(), pre_ast);
+
+        let expander = SugarExpander::new(contract_id.issuer);
+        expander.run(&mut contract_ast).unwrap();
+        assert_eq!(
+            contract_ast.expressions, ast,
+            "`foo` should have the post-comment attached"
```

Comment:
should be "end line comment" here, not "post-comment"

Reply:
Thanks! Fixed in 1495456ee.